hyper/http11/connection.py

```diff
@@ -36,6 +38,25 @@
 BODY_FLAT = 2
 
 
+def _create_tunnel(proxy_host, proxy_port, target_host, target_port,
+                   proxy_headers=None):
+    """
+    Sends CONNECT method to a proxy and returns a socket with established
+    connection to the target.
+
+    :returns: socket
+    """
+    conn = HTTP11Connection(proxy_host, proxy_port)
+    conn.request('CONNECT', '%s:%d' % (target_host, target_port),
+                 headers=proxy_headers)
+
+    resp = conn.get_response()
+    if resp.status != 200:
+        raise ProxyError("Tunnel connection failed: %d %s" % (
+            resp.status, to_native_string(resp.reason)))
```

Comment:
It might be nicer to actually return the response object in this case as well, so that users can grab it to read any response body or headers that might be there.